src/app/dataset-view/additional-components/metadata-component/components/add-polling-source/steps/data/read-form-data.ts

```diff
@@ -0,0 +1,328 @@
+                validators: {},
+                dataTestId: "enforceSchema",
+            },
+            {
+                name: "inferSchema",
+                label: "Infer schema",
+                value: false,
+                type: "checkbox",
+                tooltip:
+                    "Infers the input schema automatically from data. It requires one extra pass over the data.",
+                validators: {},
+                dataTestId: "inferSchema",
+            },
+            {
+                name: "ignoreLeadingWhiteSpace",
+                label: "Ignore leading whitespace",
+                value: false,
+                type: "checkbox",
+                tooltip:
+                    "A flag indicating whether or not leading whitespaces from values being read should be skipped.",
+                validators: {},
+                dataTestId: "ignoreLeadingWhiteSpace",
+            },
+            {
+                name: "ignoreTrailingWhiteSpace",
+                label: "Ignore trailing whitespace",
+                value: false,
+                type: "checkbox",
+                tooltip:
+                    "A flag indicating whether or not trailing whitespaces from values being read should be skipped.",
+                validators: {},
+                dataTestId: "ignoreTrailingWhiteSpace",
+            },
+            {
+                name: "multiLine",
+                label: "Multi line",
+                value: false,
+                type: "checkbox",
+                tooltip: "Parse one record, which may span multiple lines.",
+                validators: {},
+                dataTestId: "multiLine",
+            },
+        ],
+    },
+    jsonLines: {
+        controls: [
+            {
+                name: "schema",
+                label: "Schema",
+                value: "",
+                type: "schema",
+                tooltip:
+                    "A DDL-formatted schema. Schema can be used to coerce values into more appropriate data types.",
+                validators: {},
+            },
+            {
+                name: "dateFormat",
+                label: "Date format",
+                value: "",
+                type: "text",
+                tooltip: "Sets the string that indicates a date format.",
+                placeholder: "Enter date",
+                validators: {},
+            },
+            {
+                name: "encoding",
+                label: "Encoding",
+                value: "",
+                type: "text",
+                tooltip:
+                    "Allows to forcibly set one of standard basic or extended encoding.",
+                placeholder: "Enter encoding...",
+                validators: {},
+            },
+            {
+                name: "timestampFormat",
+                label: "Timestamp format",
+                value: "",
+                type: "text",
```

Comment:
Wrong type, should be a date format suggestion

Reply:
Yes, fixed on "typeahead" type.